lib/uvm/bridge.browser.js

```diff
@@ -1,142 +1,113 @@
-
-        // remove the message handler for this sandbox
-        window.removeEventListener(MESSAGE, forwardEmits);
+        // function to forward errors emitted
+        forwardErrors = function (e) {
+            bridge.emit(ERROR, e);
+        },
 
-        // do not delete sandbox element if not created for the bridge
-        !options._sandbox && iframe.parentNode && iframe.parentNode.removeChild(iframe);
-        iframe = null;
-    };
+        firmwareObjectURL,
+        worker;
 
-    // if sandbox element is provided, we simply need to init with firmware code and wait for load event to be fired
+    // if sandbox worker is provided, we simply need to init with firmware code
+    // @todo validate sandbox type or APIs
     if (options._sandbox) {
-        bridge.once('load.' + id, processCallback); // on load attach the dispatcher
-
-        iframe.contentWindow.postMessage({
-            __init_uvm: sandboxCode(code, id, true) // the last `true` param indicates firmwareOnly
-        }, TARGET_ALL);
-
-        return;
+        worker = options._sandbox;
+        worker.postMessage({ __init_uvm: firmwareCode });
     }
+    // else, spawn a new worker
+    else {
+        // convert the firmware code into a blob URL
+        firmwareObjectURL = window.URL.createObjectURL(
+            new Blob([firmwareCode], { type: 'text/javascript' })
+        );
+
+        worker = new Worker(firmwareObjectURL);
+    }
+
+    // add event listener for receiving events from worker (is removed on disconnect)
+    // don't set `onmessage` and `onerror` as it might override external sandbox
+    worker.addEventListener(MESSAGE, forwardEmits);
+    worker.addEventListener(ERROR, forwardErrors);
+
+    bridge._dispatch = function () {
+        if (!worker) {
+            return bridge.emit(ERROR,
+                new Error('uvm: unable to dispatch "' + arguments[0] + '" post disconnection.'));
+        }
+
+        worker.postMessage({
+            __emit_uvm: Flatted.stringify(Array.prototype.slice.call(arguments)),
+            __id_uvm: id
+        });
+    };
 
-    // this point onwards, it is our own iframe, so we do all the appending and other stuff to DOM
-    iframe.addEventListener(LOAD, processCallback); // removed right when executed
+    // equip bridge to disconnect (i.e. terminate the worker)
+    bridge._disconnect = function () {
+        if (!worker) { return; }
 
-    // prepare an iframe as context
-    _.forEach(sandboxAttributes, function (val, prop) {
-        iframe.setAttribute(prop, val);
-    });
+        // remove event listeners for this sandbox
+        worker.removeEventListener(MESSAGE, forwardEmits);
+        worker.removeEventListener(ERROR, forwardErrors);
 
-    // add HTML and bootstrap code to the iframe
-    iframe.setAttribute('src', 'data:text/html;base64, ' + btoa(serializedFirmware));
+        // do not terminate sandbox worker if not spawned for the bridge
+        if (!options._sandbox) {
```

Comment:
Does this activity have a chance to error out and needs to be in a try-block? Try putting infinite scripts and see what termination does to it.

Reply:
No, this immediately terminates the Worker and doesn't wait for the worker to finish its operation.